### PR TITLE
Firestore FieldValues and persistence

### PR DESCRIPTION
**Pull request by pulyaevskiy**
* Adds `Firestore.enablePersistence`
* Adds `FieldValue` class with all currently supported values: arrayUnion, arrayRemove, delete, serverTimestamp
* Adds `arrayContains` argument in `Query.where` method.

Addresses flutter/flutter#13905 and probably a bunch of other issues.

### CODE REVIEW

**Comment by pulyaevskiy** [comment]
This requires manual pod update of `Firebase/Firestore` (ios).

**Comment by pulyaevskiy** [comment]
Thanks for taking a look @kroikie ! I pushed an update with pubspec and changelog. Given nature of this update it might worth it to bump straight to `0.8.0` but I wasn't quite sure about it so put `0.7.5`. Let me know if you prefer something else, I'll update accordingly.

File: packages/cloud_firestore/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## 0.7.5
```
Comment (kroikie) [inline packages/cloud_firestore/CHANGELOG.md]: Since you are adding some new functionality, lets go with: `0.8.0`

**Comment by kroikie** [comment]
Agree to `0.8.0`.

**Comment by pulyaevskiy** [comment]
Alright, done!

**Pull request merged**